cpp/ql/lib/ext/CRegKey.model.yml

```diff
@@ -0,0 +1,19 @@
+extensions:
+  - addsTo:
+      pack: codeql/cpp-all
+      extensible: summaryModel
+    data: # namespace, type, subtypes, name, signature, ext, input, output, kind, provenance
+      - ["", "CRegKey", True, "CRegKey", "(CRegKey&)", "", "Argument[*0]", "Argument[-1]", "value", "manual"]
+      - ["", "CRegKey", True, "CRegKey", "(HKEY)", "", "Argument[0]", "Argument[-1]", "taint", "manual"]
+      - ["", "CRegKey", True, "Attach", "", "", "Argument[0]", "Argument[-1]", "taint", "manual"]
```

Comment:
I feel we should also model `CRegKey::Create`, which I believe is commonly used when adding new data structures in the registry.  The `lpszKeyName` (`Argument[*1]`) should have taint flow into the qualifier.

(it could also be a query sink, you would not want untrusted data controlling this operation)

Reply:
Agreed. I've added this summary model in 674dbce36d71d284790d6ecc1864a7887424d835. I think this is one that could also be treated with `MapKey` flow at some point and be made value-preserving.

Reply:
I don't think it will be value preserving as you're still appending the key name to an existing registry path represented by the current state of the key.  Unless maybe you consider the key to be a set of path edges, in which case I suppose you're adding one value.

Reply:
> Unless maybe you consider the key to be a set of path edges, in which case I suppose you're adding one value.

Yeah, that was my thinking. But I guess we can settle on those details later. I agree that there are multiple ways to think of this.